daemon/commit.go

```diff
@@ -20,7 +21,10 @@ type ContainerCommitConfig struct {
 // The image can optionally be tagged into a repository.
 func (daemon *Daemon) Commit(container *Container, c *ContainerCommitConfig) (*image.Image, error) {
 	if c.Pause && !container.isPaused() {
-		container.pause()
+		if err := container.pause(); err != nil && !strings.Contains(err.Error(), "is not running") && strings.Contains(err.Error(), "is already paused") {
```

Comment:
Agree, we should have an error type and maybe a function like `daemon.isNotRunning(Err)`, and `daemon.isPaused(err)`